packages/wow-ui/src/hooks/useTableCheckState.ts

```diff
@@ -13,33 +13,31 @@ const useTableCheckState = (
     }
   }, [selectedRowsProp]);
 
-  const handleRowCheckboxChange = useCallback(
-    (rowData: number) => {
-      setSelectedRows((prevSelectedRows) => {
-        const rowId = rowData;
-        const isSelected = prevSelectedRows.some((row) => row === rowId);
-        const newSelectedRows = isSelected
-          ? prevSelectedRows.filter((row) => row !== rowId)
-          : [...prevSelectedRows, rowData];
-        if (onChange) {
-          onChange(newSelectedRows);
-        }
-        return newSelectedRows;
-      });
+  const updateSelectedRow = useCallback(
+    (newSelectedRows: number[]) => {
+      if (onChange) {
+        onChange(newSelectedRows);
+        return;
+      }
+      setSelectedRows(newSelectedRows);
     },
     [onChange]
   );
 
+  const handleRowCheckboxChange = (rowData: number) => {
+    const rowId = rowData;
+    const isSelected = selectedRows.some((row) => row === rowId);
+    const newSelectedRows = isSelected
+      ? selectedRows.filter((row) => row !== rowId)
+      : [...selectedRows, rowData];
+    updateSelectedRow(newSelectedRows);
+  };
```

Comment:
p3;
여기도 selectedRows 가 많아지면 연산하는 비용이  커질 수 있을 거 같아요..!! 내부에서 selectedRows  를 Set 자료구조로 바꾸어서 해당 row 가 selected 가 선택되었는지 파악해도 좋을 거 같아요~

+) rowId 변수에 rowData 를 할당하지 않고 바로 rowData 를 넘길 수도 있을 거 같습니다!

```suggestion
      updateSelectedRow((prevSelectedRows) => {
    const newSelectedRows = new Set(prevSelectedRows);
    if (newSelectedRows.has(rowData)) {
      newSelectedRows.delete(rowData);
    } else {
      newSelectedRows.add(rowData);
    }
    return Array.from(newSelectedRows);
  });
```